src/hard_coded_unrolled_functions.jl

```diff
@@ -0,0 +1,111 @@
+@inline _unrolled_map(::Val{0}, f, itr) = ()
```

Comment:
Organizationally, I kind of like the hard-coded versions to be next to the generalized versions, is the main part of the code too difficult to navigate that way?

Reply:
Okay, I just combined them into one file called "manually_unrolled_functions.jl". I don't think it affects readability either way.